File: Data Structures & Algorithms/sort-an-array/submission-5.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
        for (int i = 1; i < nums.size(); i++ ) {
            int current = nums[i];
            int j = i-1;
            while (j >=0 and current < nums[j]) {
                nums[j+1] = nums[j];
                j--;
            }

            nums[j+1] = current;
            
        }

        return nums;
    }
};

File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-5.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:
        max_area = 0

        for i in range(len(heights)):
            l = i - 1
            r = i +1

            while l >= 0 and heights[l] >= heights[i]:
                l -= 1

            while r < len(heights) and heights[r] >= heights[i]:
                r += 1

            current = (r - l - 1) * heights[i]

            max_area = max(current, max_area)

        return max_area